packages/webpack-cli/lib/groups/resolveMode.js

```diff
@@ -1,21 +1,30 @@
 const PRODUCTION = 'production';
 const DEVELOPMENT = 'development';
 
-const resolveMode = (args) => {
+/*
+Mode priority:
+    - Mode flag
+    - Mode from config
+    - Mode form NODE_ENV
+*/
+
+const resolveMode = (args, configOptions) => {
     const {
         env: { NODE_ENV },
     } = process;
     const { mode } = args;
+    const { mode: configMode } = configOptions;
 
     let finalMode;
-    /**
-     * It determines the mode to pass to webpack compiler
-     * @returns {string} The mode
-     */
-    if (!mode && NODE_ENV && (NODE_ENV === PRODUCTION || NODE_ENV === DEVELOPMENT)) {
+
+    if (mode) {
+        finalMode = mode;
+    } else if (configMode) {
+        finalMode = configMode;
+    } else if (NODE_ENV && (NODE_ENV === PRODUCTION || NODE_ENV === DEVELOPMENT)) {
         finalMode = NODE_ENV;
     } else {
-        finalMode = mode || PRODUCTION;
+        finalMode = PRODUCTION;
```

Comment:
No need this, webpack do it itself (default is production)

Reply:
gotcha

Reply:
Need this because https://github.com/webpack/webpack-cli/issues/1678 hence added

Reply:
Okay, let's refactor it in future

Reply:
Sure sure